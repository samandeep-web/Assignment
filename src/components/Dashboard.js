import React, { useState } from 'react';
import  Calculate  from './Calculate';
import { DisplaySummary } from './DisplaySummary';
import { DisplayGraph } from './DisplayGraph';
import { calculateGraphData } from './GraphData';
export default function  Dashboard(){
  //using hooks
  const [prePayment, setPrePayment] = useState(2500);
	const [mortgageAmount, setMortgageAmount] = useState(100000);
	const [mortgagePeriod, setMortgagePeriod] = useState(25);
	const [interestRate, setInterestRate] = useState(5);

	const totalLoan = mortgageAmount - prePayment;
	const monthlyPayment = ((interestRate / 100 / 12) * totalLoan) / (1 - (Math.pow((1 + (interestRate / 100 / 12)),((0 - mortgagePeriod) * 12))));
	const payBack = monthlyPayment * 12 * mortgagePeriod;
	const totalInterestPaid = payBack - totalLoan;
	const annualPayments = calculateGraphData(totalLoan, mortgagePeriod, interestRate, monthlyPayment);
	


  return(
    <>
			<Calculate
				prePayment={prePayment}
				setPrePayment={setPrePayment}
				mortgageAmount={mortgageAmount}
				setMortgageAmount={setMortgageAmount}
				mortgagePeriod={mortgagePeriod}
				setMortgagePeriod={setMortgagePeriod}
				interestRate={interestRate}
				setInterestRate={setInterestRate}
			/>
      <DisplaySummary
        totalLoan={totalLoan}
				monthlyPayment={monthlyPayment}
				payBack={payBack}
				prePayment={prePayment}
				totalInterestPaid={totalInterestPaid}
				prePayment={prePayment}
				mortgagePeriod={mortgagePeriod}
				mortgageAmount={mortgageAmount}
      />
      <DisplayGraph
				mortgagePeriod={mortgagePeriod}
				annualPayments={annualPayments}
			/>
    </>
	);

  }

