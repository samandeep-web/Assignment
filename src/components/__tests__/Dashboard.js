import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from '../Dashboard';

describe('<Dashboard/>', () => {
  const wrapper = shallow(<Dashboard/>);
  it('should must match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});