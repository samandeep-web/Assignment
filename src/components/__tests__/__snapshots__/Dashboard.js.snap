// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> should must match the snapshot 1`] = `"<div class=\\"ui container\\"><div class=\\"ui grey inverted segment\\"><div class=\\"ui inverted form\\"><div class=\\"ui grid\\"><div class=\\"row\\"><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><a class=\\"ui tag label\\">Mortgage Amount</a></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><input type=\\"number\\" id=\\"mortgageAmount\\" min=\\"50000\\" max=\\"750000\\" step=\\"1000\\" value=\\"100000\\"/></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><span class=\\"grid__item--header\\">CA$100,000</span></div></div></div><br/><br/><div class=\\"ui three row grid\\"><div class=\\"row\\"><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><a class=\\"ui tag label\\">Pre-Deposit Amount</a></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><input type=\\"number\\" id=\\"prePayment\\" min=\\"1000\\" max=\\"150000\\" step=\\"1000\\" value=\\"2500\\"/></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><span class=\\"grid__item--header\\">CA$2,500</span></div></div></div><div class=\\"ui three row grid\\"><div class=\\"row\\"><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><a class=\\"ui tag label\\">Mortgage Period</a></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><input type=\\"range\\" class=\\"grid__item--range-slider\\" id=\\"mortgagePeriod\\" min=\\"5\\" max=\\"35\\" step=\\"1\\" value=\\"25\\"/></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><span class=\\"grid__item--header\\">25 Years</span></div></div></div><div class=\\"ui three row grid\\"><div class=\\"row\\"><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><a class=\\"ui tag label\\">Interest Rate</a></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><input type=\\"range\\" class=\\"grid__item--range-slider\\" id=\\"interestRate\\" min=\\"0.1\\" max=\\"8\\" step=\\"0.1\\" value=\\"5\\"/></div><div class=\\"fourteen wide mobile four wide tablet four wide computer column fd\\"><span class=\\"grid__item--header\\">5 %</span></div></div></div></div></div></div><div class=\\"grid__container\\"><div class=\\"ui grey message\\"><i class=\\"home icon\\"></i><div class=\\"header\\">Mortgage Summary</div><p>Over the <b>25</b> year amortization period, you will:<ul><li>have made <b>300</b> monthly (12x per year) payments of <b>CA$569 </b>.</li><li>have paid <b>2500</b> in principal.</li><li>Total Interest paid by you is <b>$73492</b></li><li>Total loan amount was:<b>$97500</b></li> </ul></p></div><table class=\\"ui grey inverted  table\\"><thead><tr><th>Category</th><th>Value</th></tr></thead><tbody><tr><td><h4 class=\\"ui image header\\"><img src=\\"https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTogAC3irWc9aKLbN8441GFxVD2SlgUhg2nDA&amp;usqp=CAU\\" class=\\"ui mini rounded image\\"/><div class=\\"content\\">Total Loan needed:<div class=\\"sub header\\"></div></div></h4></td><td>CA$97,500</td></tr><tr><td><h4 class=\\"ui image header\\"><img src=\\"https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSTytPMFhh-xFu2IjGrfNYSBC0Xi2tw3tigVQ&amp;usqp=CAU\\" class=\\"ui mini rounded image\\"/><div class=\\"content\\">Monthly Payment:<div class=\\"sub header\\"></div></div></h4></td><td>CA$569</td></tr><tr><td><h4 class=\\"ui image header\\"><img src=\\"https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTrA-Q7jgyaUSUQnKWTx_76SulVv01PBWDCtQ&amp;usqp=CAU\\" class=\\"ui mini rounded image\\"/><div class=\\"content\\">Total Payback needed:<div class=\\"sub header\\"></div></div></h4></td><td>CA$170,992</td></tr><tr><td><h4 class=\\"ui image header\\"><img src=\\"https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQ_pB-akmTVVvn5wbS5ork0iOo2JVyIyvIhyw&amp;usqp=CAU\\" class=\\"ui mini rounded image\\"/><div class=\\"content\\">Total Interest Paid:<div class=\\"sub header\\"></div></div></h4></td><td>CA$73,492</td></tr></tbody></table></div><div class=\\"ui black inverted \\"><div class=\\"ui graph-container\\"><canvas height=\\"150\\" width=\\"300\\"></canvas></div></div>"`;
